pkg/kubectl/util/service_port.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+// Lookup containerPort number by its named port name
+func LookupContainerPortNumberByName(pod api.Pod, name string) (int32, error) {
+	matchfound := false
+	namedport := int32(-1)
+	for _, ctr := range pod.Spec.Containers {
+		for _, ctrportspec := range ctr.Ports {
+			if ctrportspec.Name == name {
+				namedport = ctrportspec.ContainerPort
+				matchfound = true
+			}
+		}
+	}
+
+	if matchfound == false {
+		return namedport, fmt.Errorf("Pod '%s' does not have a named port '%s'", pod.Name, name)
+	} else {
+		return namedport, nil
+	}
+}
+
+// Lookup containerPort number from Service port number
+// It implements the handling of resolving container named port, as well as ignoring targetPort when clusterIP=None
+// It returns an error when a named port can't find a match (with -1 returned), or when the service does not
+// specify such port (with the input port number returned).
+func LookupContainerPortNumberByServicePort(svc api.Service, pod api.Pod, port int32) (int32, error) {
+	if svc.Spec.ClusterIP == "None" {
```

Comment:
nit: use the api constant

https://github.com/kubernetes/kubernetes/blob/0dda5c8a7b93517c758ac9e02287080ccc66d5cd/staging/src/k8s.io/api/core/v1/types.go#L3580-L3584

Reply:
Fixed.